src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -501,30 +482,59 @@ public override string ToString()
         {
             return $"{this.context.GrainInstance}.{this.config.StateName}";
         }
-    }
 
-    [Serializable]
-    public class LogRecord<T>
-    {
-        public T NewVal { get; set; }
-        public TransactionalResourceVersion Version { get; set; }
-    }
+        [Serializable]
+        private class Metadata
+        {
+            private static readonly JsonSerializerSettings settings = new JsonSerializerSettings
+            {
+                TypeNameHandling = TypeNameHandling.All,
+                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                DateFormatHandling = DateFormatHandling.IsoDateFormat,
+                DefaultValueHandling = DefaultValueHandling.Ignore,
+                MissingMemberHandling = MissingMemberHandling.Ignore,
+                NullValueHandling = NullValueHandling.Ignore,
+                ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
+                TypeNameAssemblyFormatHandling = TypeNameAssemblyFormatHandling.Simple,
+                Formatting = Formatting.None
+            };
+
+            [JsonIgnore]
+            public TransactionalResourceVersion HighestVersion { get; set; }
+            public string HighestVersionString
```

Comment:
This approach is a little hacky, with different properties for parsing, since Json.NET supports converters. Also, the JSON will contain methods suffixed with String, is that by design? I imagine we want something that looks nice but is still evolvable, and hence the reason to use Json.NET?
BTW, I'm fine addressing in a follow up PR after the beta (but before final)

Reply:
The purpose of this was to have some metadata that the storage layer needs only store, but has no reason to process in any other way.
There was a request that the data be formatted in json, so it is human readable for debugging purposes.
The TransactionalResourceVersion is convertible to string to have a simple, easy to store version value, and since this class has versions it was agreed that the format should be the same, hence the stream property.

My preferences deviate from what is here, and we'll continue to negotiate with users of this interface as the tech is proven out.

All of this is debatable, and still in flux.